substrate/frame/broker/src/dispatchable_impls.rs

```diff
@@ -437,4 +436,25 @@ impl<T: Config> Pallet<T> {
 		Self::deposit_event(Event::AllowedRenewalDropped { core, when });
 		Ok(())
 	}
+
+	pub(crate) fn ensure_core_availability(
+		status: &StatusRecord,
+		sale: &SaleInfoRecordOf<T>,
+	) -> Result<(), DispatchError> {
+		ensure!(sale.first_core < status.core_count, Error::<T>::Unavailable);
+		ensure!(sale.cores_sold < sale.cores_offered, Error::<T>::SoldOut);
+
+		Ok(())
+	}
+
+	/// If there is an ongoing sale returns the current price of a core.
+	pub fn api_sale_price() -> Result<BalanceOf<T>, DispatchError> {
```

Comment:
Why not just `sale_price`? I don't think we normally prefix calls with `api_`, the fact that it's available means it's in the API :).

Reply:
We already have a function called `sale_price`.

https://github.com/paritytech/polkadot-sdk/blob/cc1e6ac301ea88e3cb3253a84e4c6aa28f2d8f87/substrate/frame/nomination-pools/src/lib.rs#L3566

Reply:
Maybe even `current_price` would be more descriptive, since in other places the sale refers to the whole sale period, whereas we want the price now. This would also allow the same name to be used for the runtime api function.

Reply:
That makes sense, will update it now.